main.tf

```diff
@@ -35,7 +35,7 @@ resource "aws_vpc" "this" {
 
   assign_generated_ipv6_cidr_block     = var.enable_ipv6 && !var.use_ipam_pool ? true : null
   ipv6_cidr_block                      = var.ipv6_cidr
-  ipv6_ipam_pool_id                    = var.ipv6_ipam_pool_id
+  ipv6_ipam_pool_id                    = var.ipv6_ipam_pool_id != "" ? var.ipv6_ipam_pool_id : null
```

Comment:
> So this is a fix for us (and others), which allows null or empty strings to be passed to this module.

It already supports this, and it defaults to `null` - I'm not following why this change is warranted or what problem it is addressing

Reply:
Hey @bryantbiggs ,

It's working around this issue: https://github.com/hashicorp/terraform/issues/24142

I'm seeing this error when passing `ipv6_ipam_pool_id  = null` (as `null` and not `"null"`):

```
Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 27, in resource "aws_vpc" "this":
  27:   assign_generated_ipv6_cidr_block = var.enable_ipv6 && !var.use_ipam_pool  true : null

"assign_generated_ipv6_cidr_block": conflicts with ipv6_ipam_pool_id

Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 29, in resource "aws_vpc" "this":
  29:   ipv6_ipam_pool_id                = var.ipv6_ipam_pool_id

"ipv6_ipam_pool_id": conflicts with assign_generated_ipv6_cidr_block

time=2023-03-22T18:46:38Z level=error msg=Terraform invocation failed in /tmp/274f71cb-cc98-48bf-bfe2-5b91e9064b0a/sase-preprod/us-west-2/sase-preprod/vpc/.terragrunt-cache/7bSSPeYRzQyBwlya0Kvd1mmg3ls/m48TkmguM93MPX0LUY4qqcafT_g prefix=[/tmp/274f71cb-cc98-48bf-bfe2-5b91e9064b0a/sase-preprod/us-west-2/sase-preprod/vpc] 
time=2023-03-22T18:46:38Z level=error msg=1 error occurred:
	* exit status 1
```

Reply:
why do you need to re-pass `null` when it defaults to `null`?

Reply:
Because we're using Terragrunt to create VPCs in 5 accounts and 27 different regions...  Of those, some of those were created before we added support for BYOIPv6 so we use logic:

```
locals {
  ..

  #############################################################################
  # Extract out common variables for reuse
  #############################################################################
  ...
  ipv6_ipam_pool_id     = try(local.environment_vars.locals.vpc_ipv6_ipam_pool_id, null)
  use_ipam_pool         = local.ipv6_ipam_pool_id != null
  ...
}


inputs = {
  ...
  ipv6_ipam_pool_id                = local.ipv6_ipam_pool_id
  use_ipam_pool                    = local.use_ipam_pool
  ..
```